src/AI/agents/api/routes/websocket.py

```
@@ -0,0 +1,141 @@
+                        }, websocket)
+                    
+                    elif message_type == "session":
+                        # Register session
+                        session_id = data.get("session_id")
+                        if session_id:
+                            connection_manager.session_connections[session_id] = websocket
+
+                            # Subscribe to agent events for this session
+                            async def event_handler(event: AgentEvent):
+                                # Forward agent events to WebSocket
+                                if session_id == event.session_id:
```

Ruff (0.14.8)

Function definition does not bind loop variable `session_id`

(B023)